rotkehlchen/chain/evm/utils.py

```diff
@@ -0,0 +1,135 @@
+import logging
+from typing import TYPE_CHECKING, Any, Callable, Final, Optional
+
+from web3.types import BlockIdentifier
+
+from rotkehlchen.assets.asset import EvmToken
+from rotkehlchen.chain.evm.contracts import EvmContract
+from rotkehlchen.constants import ZERO
+from rotkehlchen.constants.resolver import evm_address_to_identifier
+from rotkehlchen.errors.asset import UnknownAsset, WrongAssetType
+from rotkehlchen.errors.misc import BlockchainQueryError, RemoteError
+from rotkehlchen.fval import FVal
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.types import VELODROME_POOL_PROTOCOL, EvmTokenKind, Price
+
+if TYPE_CHECKING:
+    from rotkehlchen.chain.evm.node_inquirer import EvmNodeInquirer
+    from rotkehlchen.types import LP_TOKEN_CONTRACT_ABI_NAMES
+
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+TOKEN_PROTOCOL_TO_ABI_NAME: Final[dict[str, 'LP_TOKEN_CONTRACT_ABI_NAMES']] = {
+    VELODROME_POOL_PROTOCOL: 'VELO_V2_LP',
+}
+
+
+def find_price_from_pool_contract(
+        evm_inquirer: 'EvmNodeInquirer',
+        token: EvmToken,
+        token_price_func: Callable,
+        token_price_func_args: list[Any],
+        block_identifier: BlockIdentifier,
+) -> Optional[Price]:
+    """
+    Calculate the price for an LP token the contract of which is also the contract of the
+    pool it represents. For example velodrome lp tokens. The price is calculated like this:
+    price = (Total value of liquidity pool) / (Current supply of LP tokens)
+    We need:
+    - Price of token 0
+    - Price of token 1
+    - Pooled amount of token 0
+    - Pooled amount of token 1
+    - Total supply of pool token
+    """
+    address = token.evm_address
+    abi_name = TOKEN_PROTOCOL_TO_ABI_NAME[token.protocol]  # the lp token's contract abi that will be used to query the info needed for lp token price calculation  # noqa: E501
+    contract = EvmContract(
+        address=address,
+        abi=evm_inquirer.contracts.abi(abi_name),
+        deployed_block=0,
+    )
+    methods = ('token0', 'token1', 'totalSupply', 'getReserves', 'decimals')
+    multicall_method = evm_inquirer.multicall
+    if isinstance(block_identifier, int) and block_identifier <= 7929876:
+        log.error(
+            f'No multicall contract at the block {block_identifier}. {abi_name} '
+            f'query failed. Should implement direct queries',
+        )
+        return None
```

Comment:
don't leave magic numbers in the code. Define as constant

```python
MULTICALL_DEPLOYMENT_BLOCK = 7929876
````

and then 

```suggestion
    if isinstance(block_identifier, int) and block_identifier <= MULTICALL_DEPLOYMENT_BLOCK:
        log.error(
            f'No multicall contract at the block {block_identifier}. {abi_name} '
            f'query failed. Should implement direct queries',
        )
        return None
```